File: go/config/WebUICfg.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// WebUICfg 对应 config/WebUICfg.py。
type WebUICfg struct {
	*session.ApiGroup
}

// NewWebUICfg 创建 WebUICfg API 分组。
func NewWebUICfg(s *session.Session) *WebUICfg {
	return &WebUICfg{ApiGroup: session.NewApiGroup(s)}
}

// Config 对应 webuicfg/config.xml。
func (w *WebUICfg) Config() (map[string]interface{}, error) {
	return w.S.Get("webuicfg/config.xml", nil, "config")
}

File: src/config/web_ui_cfg.rs
//! WebUICfg config group (`config/WebUICfg.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// WebUICfg config group.
pub struct WebUICfgConfig<'a> {
    conn: &'a Connection,
}

impl<'a> WebUICfgConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        WebUICfgConfig { conn }
    }

    /// `webuicfg/config.xml`. Web UI configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/webuicfg/config.xml")
    }
}
